custom_components/fellow/fellow_aiden/__init__.py

```
@@ -315,28 +331,23 @@ def create_schedule(self, data):
-        if 'id' not in parsed:
-            message = parsed.get('message', 'Unable to get error message.')
-            if 'Profile could not be found' in message:
+        response = self._request_with_reauth("post", schedule_url, json=data)
+        self._ensure_success(response, "Schedule creation")
+
+        parsed = self._parse_response(response)
+        if "id" not in parsed:
+            message = parsed.get("message", "Unable to get error message.")
+            if "Profile could not be found" in message:
                 message += "Valid profiles: %s" % self.__get_profile_ids()
             raise Exception("Error in processing: %s" % message)
```

Ruff (0.15.1)

[warning] Create your own exception

(TRY002)